spec/helpers/date_helper_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+RSpec.describe DateHelper do
+  describe '.format_date_of_birth' do
+    it 'formats a date from a date' do
+      expect(helper.format_date_of_birth(Date.parse('2014-07-24'))).to eq('24 July 2014')
+    end
+    it 'formats a date from a string' do
+      expect(helper.format_date_of_birth('2014-07-24')).to eq('24 July 2014')
+    end
+  end
+
+  describe '.format_date_of_visit' do
+    it 'formats a day from a string' do
+      expect(helper.format_date_of_visit('2014-07-24')).to eq('Thursday 24 July')
+    end
+    it 'formats a day from a date' do
+      expect(helper.format_date_of_visit(Date.parse('2014-07-24'))).to eq('Thursday 24 July')
+    end
+  end
+
+  describe '.format_time_str' do
+    it 'formats a time from a string' do
+      expect(helper.format_time_str('0900')).to eq('9:00am')
+      expect(helper.format_time_str('1200')).to eq('12:00pm')
+      expect(helper.format_time_str('1545')).to eq('3:45pm')
+    end
+  end
+
+  describe '.format_time_str_24' do
+    it 'formats a time in 24 clock from a string' do
+      expect(helper.format_time_str_24('0900')).to eq('09:00')
+      expect(helper.format_time_str_24('1200')).to eq('12:00')
+      expect(helper.format_time_str_24('1545')).to eq('15:45')
+    end
+  end
+
+  describe '.display_start_time' do
+    it 'formats a start time from a 24hr time string' do
+      expect(helper.display_start_time('0945')).to eq('9:45am')
+    end
+  end
+
+  describe '.display_slot_and_duration' do
+    it 'displays a slot and duration' do
+      expect(helper.display_slot_and_duration('0945-1145')).to eq('9:45am for 2 hrs')
+      expect(helper.display_slot_and_duration('1430-1600')).to eq('2:30pm for 1 hr 30 mins')
```

Comment:
Do we know how these abbreviations are handled (particularly by screen readers)? I suspect not very well…